pkg/endpoint/endpoint_test.go

```diff
@@ -171,6 +172,44 @@ func (s *EndpointSuite) TestEndpointStatus(c *C) {
 	c.Assert(eps.String(), Equals, "OK")
 }
 
+func (s *EndpointSuite) TestEndpointUpdateLabels(c *C) {
+	e := Endpoint{
+		ID:     IPv6Addr.EndpointID(),
+		IPv6:   IPv6Addr,
+		IPv4:   IPv4Addr,
+		Status: NewEndpointStatus(),
+		OpLabels: pkgLabels.OpLabels{
+			Custom:                pkgLabels.Labels{},
+			Disabled:              pkgLabels.Labels{},
+			OrchestrationIdentity: pkgLabels.Labels{},
+			OrchestrationInfo:     pkgLabels.Labels{},
+		},
+	}
+	e.Mutex.Lock()
+	e.SetDefaultOpts(nil)
+	e.Mutex.Unlock()
+
+	// Test that inserting identity labels works
+	rev := e.replaceIdentityLabels(pkgLabels.Map2Labels(map[string]string{"foo": "bar", "zip": "zop"}, "cilium"))
+	c.Assert(rev, Not(Equals), 0)
+	c.Assert(string(e.OpLabels.OrchestrationIdentity.SortedList()), Equals, "cilium:foo=bar;cilium:zip=zop;")
+	// Test that nothing changes
+	rev = e.replaceIdentityLabels(pkgLabels.Map2Labels(map[string]string{"foo": "bar", "zip": "zop"}, "cilium"))
+	c.Assert(rev, Equals, 0)
+	c.Assert(string(e.OpLabels.OrchestrationIdentity.SortedList()), Equals, "cilium:foo=bar;cilium:zip=zop;")
+	// Remove one label, change the source and value of the other.
+	rev = e.replaceIdentityLabels(pkgLabels.Map2Labels(map[string]string{"foo": "zop"}, "ngnix"))
```

Comment:
`nginx` is misspelled ;-)

Reply:
@jrajahalme Could you still fix this before merge? Thx

Reply:
Fixed.